test/functional/insert_tests.js

```diff
@@ -2886,4 +2887,31 @@ describe('Insert', function() {
       });
     }
   });
+
```

Comment:
I would like to see a unit test file that validates all of the behavior we specified in the design document yesterday. It's fine to see it in action here, but prefer tests specific to the job of creating validators and validating different types of objects.